genesyscloud/task_management_workitem_schema/data_source_genesyscloud_task_management_workitem_schema.go

```diff
@@ -0,0 +1,48 @@
+package task_management_workitem_schema
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+)
+
+/*
+   The data_source_genesyscloud_task_management_workitem_schema.go contains the data source implementation
+   for the resource.
+*/
+
+// dataSourceTaskManagementWorkitemSchemaRead retrieves by name the id in question
+func dataSourceTaskManagementWorkitemSchemaRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getTaskManagementProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+
+	// Query for workitem schemas by name. Retry in case new schema is not yet indexed by search.
+	// As schema names are non-unique, fail in case of multiple results.
+	return gcloud.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
+		schemas, retryable, err := proxy.getTaskManagementWorkitemSchemasByName(ctx, name)
+		if err != nil && !retryable {
+			return retry.NonRetryableError(fmt.Errorf("error getting workitem schema %s: %v", name, err))
+		}
+
+		if retryable {
+			return retry.RetryableError(fmt.Errorf("no workitem schema found with name %s", name))
+		}
+
+		if len(*schemas) > 1 {
```

Comment:
In the actual function call to retrieve the records we match on name.  We should probably check with the API owners and see if it is possible to have multiple schemas with the same name.  If it, then this error makes sense otherwise it seems a bit redundant.

Reply:
Hi John, yes it is possible to have multiple schemas with the same name and so I added that code:
![image](https://github.com/MyPureCloud/terraform-provider-genesyscloud/assets/22146579/5ac0847a-722f-49db-b079-117df9af74ec)